etcdserver/apply.go

```diff
@@ -257,15 +263,60 @@ func (a *applierV3backend) Range(txn mvcc.TxnRead, r *pb.RangeRequest) (*pb.Rang
 		limit = limit + 1
 	}
 
-	ro := mvcc.RangeOptions{
-		Limit: limit,
-		Rev:   r.Revision,
-		Count: r.CountOnly,
-	}
+	var rr *mvcc.RangeResult
+	var err error
+	rangeEnd := mkGteRange(r.RangeEnd)
 
-	rr, err := txn.Range(r.Key, mkGteRange(r.RangeEnd), ro)
-	if err != nil {
-		return nil, err
+	if txn != nil {
+		ro := mvcc.RangeOptions{
+			Limit: limit,
+			Rev:   r.Revision,
+			Count: r.CountOnly,
+		}
+
+		rr, err = txn.Range(r.Key, rangeEnd, ro)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		rr = &mvcc.RangeResult{
+			KVs: make([]mvccpb.KeyValue, 0),
+			Rev: r.Revision,
+		}
+
+		lim := int64(a.s.Cfg.RangeMaxKeysOnce)
+		if lim == 0 || r.Limit != 0 && r.Limit < lim {
+			lim = r.Limit
+		}
+		startKey := r.Key
+		noEnd := bytes.Compare(rangeEnd, []byte{0}) != 0
```

Comment:
Can we add simple `bytes.Compare` documentation?

e.g. `// rangeEnd == 0-byte`

Also for `bytes.Compare(startKey, rangeEnd) == -1`

Reply:
What does `byte-0` mean?

Reply:
I was saying we should document what this `bytes.Compare` does. `bytes.Compare(rangeEnd, []byte{0}) != 0` is clear since we define as `noEnd`. Can we godoc `bytes.Compare(startKey, rangeEnd) == -1`?